tests/e2e/vmc_csi_deployments.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	//dep "k8s.io/kubernetes/test/e2e/framework/deployment"
+
+	fdep "k8s.io/kubernetes/test/e2e/framework/deployment"
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+)
+
+var _ = ginkgo.Describe("[vmc-gc] [csi-guest] Deploy Update and Scale Deployments", func() {
+	f := framework.NewDefaultFramework("e2e-vsphere-deployment")
+	var (
+		namespace         string
+		client            clientset.Interface
+		storagePolicyName string
+		scParameters      map[string]string
+		replica           int32
+		maxReplica        int32
+		pvclaims          []*v1.PersistentVolumeClaim
+		pvclaim           *v1.PersistentVolumeClaim
+		sc                *storagev1.StorageClass
+		svNamespace       string
+	)
+	ginkgo.BeforeEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		namespace = getNamespaceToRunTests(f)
+		client = f.ClientSet
+		bootstrap()
+
+		ginkgo.By("Set Resource Quota")
+		svcClient, svNamespace = getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, rqLimit)
+
+		sc, err := client.StorageV1().StorageClasses().Get(ctx, storageclassname, metav1.GetOptions{})
+		if err == nil && sc != nil {
+			gomega.Expect(client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))).NotTo(gomega.HaveOccurred())
+		}
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+	})
+
+	ginkgo.AfterEach(func() {
+		ginkgo.By("Modify Resource Quota")
+		svcClient, svNamespace := getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, defaultrqLimit)
+
+		if pvclaim != nil {
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
wait for cns volume to be deleted
Also, delete deployment here, if it was not deleted in test case